git_machete/git_operations.py

```diff
@@ -580,8 +580,10 @@ def get_reflog(self, branch: AnyBranchName) -> List[GitReflogEntry]:
                                                          ))
             return self.__reflogs_cached[branch]
 
-    def create_branch(self, branch: LocalBranchShortName, out_of_revision: AnyRevision) -> None:
+    def create_branch(self, branch: LocalBranchShortName, out_of_revision: AnyRevision, switch_branch: bool) -> None:
         self._run_git("branch", branch, out_of_revision)
+        if switch_branch:
+            self._run_git("switch", branch)
```

Comment:
Note that `git switch` has only been added in git 2.23 or so

Reply:
Hmm, in that case having 2 separate functions could be better.

Reply:
As in, you can just use `git checkout <branch>` instead!

Reply:
You mean depending on git version i can do different things?

Reply:
Nopes!
Prior to 2.23: `git checkout <branch>` is the sole command for switching current branch
2.23 and beyond: both `git checkout <branch>` and `git switch <branch>` do **the same thing**

That's just it

Reply:
Just a matter of git CLI changing... but the capabilities if it comes to switching branches are the same

Reply:
@PawelLipski I am still not sure what you are proposing :D Are you proposing to use git checkout / switch depending on git version?

Reply:
Nope, just:
```suggestion
            self._run_git("checkout", branch)
```
This way we're compatible with every git version in existence! as opposed to `git switch`, which is git 2.23+ only

Reply:
Ok, i think i finally got it - i can just use `checkout` instead of `switch` since it does the same thing :D